drivers/flash/flash_rpc/flash_rpc_host.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <drivers/flash/flash_rpc.h>
+
+#include <nrf_rpc_cbor.h>
+#include <nrf_rpc/nrf_rpc_ipc.h>
+
+#include <zcbor_common.h>
+#include <zcbor_decode.h>
+#include <zcbor_encode.h>
+
+
+LOG_MODULE_DECLARE(FLASH_RPC, CONFIG_FLASH_RPC_LOG_LEVEL);
+
+#define CBOR_BUF_FLASH_MSG_SIZE (sizeof(void *) + sizeof(size_t) + sizeof(off_t))
+
+NRF_RPC_IPC_TRANSPORT(flash_rpc_api_tr, DEVICE_DT_GET(DT_NODELABEL(ipc0)), "flash_rpc_api_ept");
+NRF_RPC_GROUP_DEFINE(flash_rpc_api, "flash_rpc_api", &flash_rpc_api_tr, NULL, NULL, NULL);
+
+static const struct device *const flash_controller =
+	DEVICE_DT_GET_OR_NULL(DT_CHOSEN(zephyr_flash_controller));
```

Comment:
I am not sure about the layout of memory for the NET code but shouldn't we rather limit access to specific partition? We seem to allow write anywhere on that thing.

Reply:
So, since the bootloader protects itself, doing a flash write to it would be a nop/cause a failure. 
We should not limit writing to the rest of the device.


This is the layout for the Netcore flash controller. Are you suggesting that we target `flash1`? Or did I misunderstand? If you wanted it to use the part of the partition here(which doesn't make sense in my mind as none of these partitions is used on the network core so far).
```DTS
	partitions {
			flash_controller: flash-controller@41080000 {
			compatible = "nordic,nrf53-flash-controller";
			reg = < 0x41080000 0x1000 >;
			partial-erase;
			#address-cells = < 0x1 >;
			#size-cells = < 0x1 >;
			flash1: flash@1000000 {
				compatible = "soc-nv-flash";
				erase-block-size = < 0x800 >;
				write-block-size = < 0x4 >;
				reg = < 0x1000000 0x40000 >;
				partitions {
					compatible = "fixed-partitions";
					#address-cells = < 0x1 >;
					#size-cells = < 0x1 >;
					boot_partition: partition@0 {
						label = "mcuboot";
						reg = < 0x0 0xc000 >;
					};
					slot0_partition: partition@c000 {
						label = "image-0";
						reg = < 0xc000 0x17000 >;
					};
					slot1_partition: partition@23000 {
						label = "image-1";
						reg = < 0x23000 0x17000 >;
					};
					storage_partition: partition@3a000 {
						label = "storage";
						reg = < 0x3a000 0x6000 >;
					};
				};
			};
		};
```

Reply:
I was asking because as far as I can tell we will be using the flash driver also in MCUmgr right?

Reply:
I need to go back there - VNFD should allow full access in boot stage (B0N's VNFD instance) and limits access in application run-time stage ( networking stack VNFD instance). Maybe we should do that in follow-up patch.